### PR TITLE
feat: add wallet workflow

### PR DESCRIPTION
**Pull request by filoozom**
**Todo:**

`AccountPublicWallet`:

- [x] Visual change when the public address was copied to the clipboard
- [x] Password input modal to show the private key

`AccountWallet`:

- [x] Show xDAI balance
- [x] Add function to do the native token transfer
- [x] Redirect when the profile doesn't exist

**Out of scope:**

- Show past transactions (but I don't think that's doable without a centralized API right now)

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		name: 'xDAI',
+		symbol: 'xDAI',
+		decimals: 18,
+	},
+	rpcUrls: {
+		websocket: 'wss://rpc.gnosischain.com/wss',
+		default: 'https://rpc.ankr.com/gnosis',
+	},
+	blockExplorers: {
+		etherscan: BLOCK_EXPLORER,
```
Comment (vojtechsimetka) [inline src/config/index.ts:23]:
```suggestion
```
Do we need to provide etherscan if we set default?

File: src/components/modals/password/password.module.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.password-modal {
+	padding-top: 137px;
+	backdrop-filter: blur(10px);
+}
+
+.password-modal :global .modal-content {
+	background-color: transparent;
```
Comment (vojtechsimetka) [inline src/components/modals/password/password.module.css:7]: This needs to have some color.
Screenshot 2022-06-01 at 18 36 48

File: src/pages/user-wallet/wallet.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+				{/* eslint-disable-next-line @typescript-eslint/no-explicit-any */}
+				{isError && (error as any).reason}
+				{isLoading && 'Loading...'}
+				{!isValid && amount && address && 'Form invalid'}
+
+				<div class="btns btn-icons">
+					<Link to={ACCOUNT_WALLET} className="close">
+						<img src={cancelButton} />
+					</Link>
```
Comment (vojtechsimetka) [inline src/pages/user-wallet/wallet.tsx:103]: Can use `ButtonClose`

File: src/pages/user-wallet/wallet.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+					<a role="button" type="submit" class="btn-icon" onClick={submit}>
+						<img src={sendButton} />
+					</a>
```
Comment (vojtechsimetka) [inline src/pages/user-wallet/wallet.tsx:106]: Should be `ButtonRoundArrow`

**Review by vojtechsimetka — APPROVED** [review]
generally LGTM.

I was testing this and I wondered if the next step should be to be able to spawn local blockchain.

File: src/pages/user-wallet/public.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+							</div>
+						</>
+					) : (
+						<div class="btns">
+							<button
+								class="btn btn-light"
+								onClick={() => setShowPassword(true)}
+							>
+								show private key
+							</button>
```
[the remark this answers is not in the document]
Comment (filoozom) [reply src/pages/user-wallet/public.tsx:98]: For the "show private key" button?

File: src/components/modals/password/password.module.css [hunk 6]
```diff
@@ -0,0 +1,16 @@
+.password-modal {
+	padding-top: 137px;
+	backdrop-filter: blur(10px);
```
[the remark this answers is not in the document]
Comment (vojtechsimetka) [reply src/components/modals/password/password.module.css:3]: It does not, actually we don't need it I guess, I am just used to having z-index in modals but the default value should be good.

**Pull request merged**